### PR TITLE
CURA-10951_gh_build_curapackage

### PR DESCRIPTION
**Pull request by jellespijker**
On Windows the pyprojecttoolchain used the the build folder as output for the sip-generated files, this folder was created before setuptools.py ran by conan. By adding a specific sip folder and changing the SIP CMake macros to use that folder. setuptools should generate the sip c files and headers and cmake should builds the binaries from those files.

The two updated recipes are already in use btw: with `conan export-pkg .` and a `conan upload ... -r cura` which needs to be done by hand anyhow in this repo, since we don't have CI/CD setup yet.

### CODE REVIEW

File: recipes/pyprojecttoolchain/conanfile.py [hunk 1]
```diff
@@ -69,21 +69,54 @@ def context(self):
...
         }
 
 
+class ToolSipProjectPyQtBuilder(Block):
+    template = textwrap.dedent("""
+    {% if link_full_dll %}link-full-dll = true
+    {% endif %}py-pylib-dir = "{{ py_pylib_dir }}"
+    py-pylib-lib = "{{ py_pylib_lib }}"
+    {% if py_pylib_shlib is not none %}py-pylib-shlib = "{{ py_pylib_shlib }}"
+    {% endif %}qml-debug = {{ qml_debug | lower }}
```
Comment (casperlamboo) [inline recipes/pyprojecttoolchain/conanfile.py:83]:
```suggestion
    {% if link_full_dll %}link-full-dll = true{% endif %}
    py-pylib-dir = "{{ py_pylib_dir }}"
    py-pylib-lib = "{{ py_pylib_lib }}"
    {% if py_pylib_shlib is not none %}py-pylib-shlib = "{{ py_pylib_shlib }}"{% endif %}
    qml-debug = {{ qml_debug | lower }}
```
Any reason why we don't do it like this? if, and end ifs on the same line for single line changes makes it more readable imo

File: recipes/pyprojecttoolchain/conanfile.py [hunk 2]
```diff
@@ -69,21 +69,54 @@ def context(self):
...
+        }
+
+
 class ToolSipProjectBlock(Block):
     template = textwrap.dedent("""
     [tool.sip.project]
     compile = {{ compile | lower }}
-    sip-files-dir = "{{ sip_files_dir }}"
+    {% if sip_files_dir is not none %}sip-files-dir = "{{ sip_files_dir }}"
+    {% endif %}
```
Comment (casperlamboo) [inline recipes/pyprojecttoolchain/conanfile.py:114]:
```suggestion
    {% if sip_files_dir is not none %}sip-files-dir = "{{ sip_files_dir }}"{% endif %}
```

File: recipes/pyprojecttoolchain/conanfile.py (hunk 1 above)
Comment (jellespijker) [reply recipes/pyprojecttoolchain/conanfile.py:83]: I'm fine with the suggested change if the generated toolchain still jas the correct format with respect to new lines

File: recipes/pyprojecttoolchain/conanfile.py (hunk 2 above)
Comment (jellespijker) [reply recipes/pyprojecttoolchain/conanfile.py:114]: See reply above

File: recipes/pyprojecttoolchain/conanfile.py (hunk 1 above)
Comment (rburema) [reply recipes/pyprojecttoolchain/conanfile.py:83]: Let's try this _after_ merge, so we know this wasn't the culprit when something goes wrong.

File: recipes/pyprojecttoolchain/conanfile.py (hunk 2 above)
Comment (rburema) [reply recipes/pyprojecttoolchain/conanfile.py:114]: Ditto.

**Review by rburema — APPROVED** [review]

**Pull request merged**